src/building/building_state.c

```diff
@@ -532,6 +534,11 @@ void building_state_load_from_buffer(buffer *buf, building *b, int building_buf_
         b->fumigation_direction = buffer_read_u8(buf);
     }
 
+    if (save_version <= SAVE_GAME_CARAVANSERAI_LIGHTHOUSE_WRONG_ID && (b->type == BUILDING_LIGHTHOUSE || b->type == BUILDING_CARAVANSERAI)) {
```

Comment:
If we change this to:

```
if (b->figure_id2 && figure_get(b->figure_id2)->type != FIGURE_LABOR_SEEKER)
```

We won't need to bump the savegame version while also fixing the problem for old saves.

Reply:
Ok...I changed it around but I had to add a couple of checks. With the new saved game previews, figure_get() can't be used when loading the game because it's either uninitialized on the main menu (causing the game to crash) or is pointing to your current city, so I put a flag in to let the function know if we're loading in that state. The other issue is market traders use figure_id2 so this would move them to the wrong spot. Anyways, let me know if you like it better this way.

Reply:
Very good point. Looks good now!